packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx

```diff
@@ -46,60 +52,48 @@ const PasswordPanel: FC = () => {
         }
     }, [error, showError]);
 
-    const handleUpdate = () => {
-        if (hasPassword && password && newPassword) {
-            mutate({
-                password,
-                newPassword,
-            });
-        } else if (!hasPassword && newPassword) {
-            mutate({
-                password: '',
-                newPassword,
-            });
-        } else {
-            showToastError({
-                title: 'Required fields: password and new password',
-                timeout: 3000,
-            });
-        }
-    };
+    const onSubmit = form.onSubmit(({ currentPassword, newPassword }) => {
+        updateUserPassword({
+            password: currentPassword,
+            newPassword,
+        });
+    });
 
     return (
-        <div
-            style={{
-                height: 'fit-content',
-                display: 'flex',
-                flexDirection: 'column',
-            }}
-        >
-            {hasPassword && (
+        <Flex dir="column" sx={{ height: 'fit-content' }}>
+            <form onSubmit={onSubmit} style={{ width: '100%' }}>
+                {hasPassword && (
+                    <PasswordInput
+                        label="Current password"
+                        placeholder="Enter your password..."
+                        required
+                        disabled={isLoading}
+                        {...form.getInputProps('currentPassword')}
+                    />
+                )}
                 <PasswordInput
-                    label="Current password"
-                    placeholder="Enter your password..."
+                    label="New password"
+                    placeholder="Enter your new password..."
                     required
                     disabled={isLoading}
-                    value={password}
-                    onChange={setPassword}
+                    mt="md"
+                    {...form.getInputProps('newPassword')}
                 />
-            )}
-            <PasswordInput
-                label="New password"
-                placeholder="Enter your new password..."
-                required
-                disabled={isLoading}
-                value={newPassword}
-                onChange={setNewPassword}
-            />
-            <div style={{ flex: 1 }} />
-            <Button
-                style={{ alignSelf: 'flex-end', marginTop: 20 }}
-                intent={Intent.PRIMARY}
-                text="Update"
-                onClick={handleUpdate}
-                loading={isLoading}
-            />
-        </div>
+
+                <Button
+                    mt="md"
+                    type="submit"
+                    ml="auto"
+                    display="flex"
```

Comment:
no need for flex

Reply:
changed to `display='block'` similar to comment on: https://github.com/lightdash/lightdash/pull/5326#discussion_r1183341274